modules/tinymce/src/themes/silver/test/ts/module/CommonMailMergeFetch.ts

```diff
@@ -0,0 +1,107 @@
+import { TestHelpers } from '@ephox/alloy';
+import { Toolbar } from '@ephox/bridge';
+import { Arr } from '@ephox/katamari';
+
+export const fetchMailMergeData = (settings: { collapseSearchResults: boolean }, store: TestHelpers.TestStore): Toolbar.ToolbarMenuButtonSpec['fetch'] => (callback, fetchContext) => {
+  const makeMailMerge = (info: { value: string; title?: string}) => ({
+    type: 'menuitem',
+    text: info.title ?? info.value,
+    onAction: () => {
+      store.adder('Triggering: ' + info.value)();
+    }
+  });
+
+  const makeCategory = (title: string, items: any[]) => ({
+    type: 'nestedmenuitem',
+    text: title,
+    getSubmenuItems: () => items
+  });
+
+  const currentDateMerge = {
+    value: 'Current.Date',
+    title: 'Current date in DD/MM/YYYY format'
+  };
+
+  const tocMerge = {
+    value: 'Campaign.Toc',
+    title: 'Linked table of contents in your campaign'
+  };
+
+  const phoneHomeMerge = { value: 'Phone.Home' };
+  const phoneWorkMerge = { value: 'Phone.Work' };
+
+  const personFirstnameMerge = { value: 'Person.Name.First' };
+  const personSurnameMerge = { value: 'Person.Name.Last' };
+  const personFullnameMerge = { value: 'Person.Name.Full' };
+
+  const personWorkEmail = { value: 'Person.Email.Work' };
+  const personHomeEmail = { value: 'Person.Email.Home' };
+
+  const hasNoSearchPattern = !fetchContext || (fetchContext && fetchContext.pattern.length === 0);
+
+  if (!settings.collapseSearchResults || hasNoSearchPattern) {
+    callback([
+      makeMailMerge(currentDateMerge),
+      makeMailMerge(tocMerge),
+      makeCategory(
+        'Phone',
+        [
+          makeMailMerge(phoneHomeMerge),
+          makeMailMerge(phoneWorkMerge)
+        ]
+      ),
+      makeCategory(
+        'Person',
+        [
+          makeMailMerge(personFirstnameMerge),
+          makeMailMerge(personSurnameMerge),
+          makeMailMerge(personFullnameMerge),
+          makeCategory(
+            'Email',
+            [
+              makeMailMerge(personWorkEmail),
+              makeMailMerge(personHomeEmail)
+            ]
+          )
+        ]
+      )
+    ] as any);
```

Comment:
Likewise here? Using `any` here just makes me worried that when things change later this would be missed.

Reply:
- [ ] use types